src/main/java/com/github/junrar/rarfile/FileHeader.java

```diff
@@ -642,6 +648,6 @@ public boolean isDirectory() {
      * @since 7.2.0
      */
     public String getFileName() {
-        return isUnicode() ? fileNameW : fileName;
+        return isUnicode() && fileNameW != null && !fileNameW.isEmpty() ? fileNameW : fileName;
```

Comment:
I suppose this is somehow related, as we can have `hasExtTime` while not having actual extended time data, we can have `isUnicode` while not having actual unicode filename ?

Reply:
Yes, we can, in theory. If you check the bit of code that fills those fields, you can see `isUnicode()` can be true, but if the length of the decoded string is what's expected, `fileNameW` can be set to an empty string.

Reply:
https://github.com/junrar/junrar/blob/622891d137975adab6952fa344096b6ad3955b7c/src/main/java/com/github/junrar/rarfile/FileHeader.java#L151